fastpurge/_client.py

```diff
@@ -32,6 +36,29 @@
 ])
 
 
+class LoggingRetry(Retry):
+    def __init__(self, *args, **kwargs, ):
+        self._logger = kwargs.pop('logger', None)
+        super(LoggingRetry, self).__init__(*args, **kwargs)
+
+    def new(self, **kw):
+        kw['logger'] = self._logger
+        return super(LoggingRetry, self).new(**kw)
+
+    def increment(self, method, url, *args, **kwargs):
+        response = kwargs.get("response")
+        if response:
+            self._logger.error("An invalid status code %s was received "
+                               "when trying to %s to %s: %s"
+                               % (response.status, method, url, response.reason))
```

Comment:
Minor: the preferred logging style is to let the logger itself do the interpolation, e.g.

```suggestion
            self._logger.error("An invalid status code %s was received "
                               "when trying to %s to %s: %s",
                               response.status, method, url, response.reason)
```

One reason is that any exceptions while converting the arguments to strings will themselves then be gracefully logged instead of just crashing.